### PR TITLE
Add release information to documentation

### PR DESCRIPTION
**Pull request by forsyth2**
Add information about releases on GitHub and Anaconda to documentation.

### CODE REVIEW

**Comment by forsyth2** [comment]
If anything else needs to be updated in #56, those steps should be added to these docs too.

File: source/release.rst [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        git commit -am 'Update version numbers'
+        git push origin master
+
+Or:
+
+    ::
+
+        git commit -am 'Update version numbers'
+        git push <fork-name> <branch-name>
+        # Create pull request for the gh-pages branch
```
Comment (forsyth2) [inline source/release.rst]: #`master` not `gh-pages`

File: docs/html/_sources/release.rst.txt [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        git checkout -b <branch-name> origin/master
+
+2. Edit ``version`` in ``setup.py`` to the new version.
+Don't prefix this with a "v".
+
+3. Edit ``__version__`` in ``zstash/__init__.py``. Prefix this with a "v".
+
+5. Change the ``version``  in ``conda/meta.yaml``.
+``version`` is what the version of the software will be on Anaconda.
+Don't prefix this with a "v".
```
Comment (xylar) [inline docs/html/_sources/release.rst.txt]:
```suggestion
5. Change the ``version`` and ``build``  in ``conda/meta.yaml``.
``version`` is what the version of the software will be on Anaconda.
Don't prefix this with a "v".  Reset the ``build`` to 0 if necessary (i.e. if higher builds of the 
previous version have been made).
```

File: docs/html/_sources/release.rst.txt [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+1. Go to the Releases on the GitHub repo of the project
+`here <https://github.com/E3SM-Project/zstash/releases>`_.
+and draft a new release.
+
+2. ``Tag version`` and ``Release title`` should both be the version, including the "v".
+``Target`` should be ``master``. Use ``Describe this release`` to write what features
+the release adds. You can scroll through
+`Zstash commits <https://github.com/E3SM-Project/zstash/commits/master>`_ to see
+what features have been added recently.
+
```
Comment (xylar) [inline docs/html/_sources/release.rst.txt]: Perhaps here is where to add 3. on downloading the tar.gz file, getting the `sha256` checksum, and making a PR to update it in the `conda/meta.yaml`

**Pull request merged**